dotCMS/src/main/java/com/dotmarketing/common/db/DBTimeZoneCheck.java

```diff
@@ -0,0 +1,73 @@
+package com.dotmarketing.common.db;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Timestamp;
+import java.util.TimeZone;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.dotcms.util.CloseUtils;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Logger;
+import com.zaxxer.hikari.HikariDataSource;
+import io.vavr.control.Try;
+
+public class DBTimeZoneCheck {
+    private static final AtomicReference<Class> DRIVER_REF = new AtomicReference<>(null);
+
+    /**
+     * Driver class loaded to avoid call the loading method.
+     * @param dataSource
+     * @return
+     * @throws ClassNotFoundException
+     */
+    private static Class resolveDriverClass(final HikariDataSource dataSource) throws ClassNotFoundException {
+        DRIVER_REF.compareAndSet(null, Class.forName(dataSource.getDriverClassName()));
+        return DRIVER_REF.get();
+    }
+
+    /**
+     * This code checks to see if the timezone id being passed in is a valid id both with java and with
+     * the db
+     * 
+     * @param timezone
+     * @return
+     */
+    public static boolean isTimeZoneValid(final String timezone) {
+        Connection connection = null;
+        PreparedStatement statement = null;
+        ResultSet resultSet = null;
+        final TimeZone defaultTimeZone = TimeZone.getDefault();
+        try {
+            final HikariDataSource hikari = (HikariDataSource) DbConnectionFactory.getDataSource();
+            resolveDriverClass(hikari);
+
+            final TimeZone testingTimeZone = TimeZone.getTimeZone(timezone);
+            if (!testingTimeZone.getID().equalsIgnoreCase(timezone)) {
+                throw new DotRuntimeException("Invalid Timezone: " + timezone);
+            }
+            TimeZone.setDefault(testingTimeZone);
+
+            connection = DriverManager.getConnection(hikari.getJdbcUrl(), hikari.getUsername(), hikari.getPassword());
+
+            statement = connection.prepareStatement("SELECT * FROM inode WHERE idate > ?");
```

Comment:
why not use DotConnect??

Reply:
Using `DotConnect` we cannot if the recent default Tiemzone set was going to lead to a failure from the driver.